packages/ckeditor5-watchdog/src/editorwatchdog.ts

```diff
@@ -165,12 +164,24 @@ export default class EditorWatchdog<TEditor extends Editor = Editor> extends Wat
 				console.error( 'An error happened during the editor destroying.', err );
 			} )
 			.then( () => {
+				const existingRoots = Object.keys( this._data!.roots ).reduce( ( acc, rootName ) => {
+					acc[ rootName ] = '';
+
+					return acc;
+				}, {} as Record<string, string> );
+
+				const updatedConfig = {
+					...this._config,
+					plugins: this._config!.plugins || [],
+					_watchdogInitialData: this._data
+				};
+
+				updatedConfig.plugins!.push( EditorWatchdogInitPlugin as any );
```

Comment:
I would push it to `extraPlugins`. AFAIR, if you `config.plugins` is not set, then setting it to `[ WatchdogInitPlugin ]` will result in an editor that has no plugins. And it is possible to leave `config.plugins` empty if you use editor build. So, do the same, but use `extraPlugins`.